### PR TITLE
`convert_na_to` improvements when dealing with character

### LINKED ISSUE #214: `convert_na_to` improvements when dealing with character <NA>?
Reprex:

``` r
library(datawizard)
df <- data.frame(letters, numbers = 1:26, factors = as.factor(letters))[1:5,]
df[2:3,] <- NA
df
#>   letters numbers factors
#> 1       a       1       a
#> 2    <NA>      NA    <NA>
#> 3    <NA>      NA    <NA>
#> 4       d       4       d
#> 5       e       5       e

convert_na_to(df, replacement = 0)
#>   letters numbers factors
#> 1       a       1       a
#> 2    <NA>      NA    <NA>
#> 3    <NA>      NA    <NA>
#> 4       d       4       d
#> 5       e       5       e
# I think this should work when you want to convert numeric, character, 
# and factor NAs all to the same value...

convert_na_to(df, replace_num = 0,
              replace_char = 0,
              replace_fac = 0)
#>   letters numbers factors
#> 1       a       1       a
#> 2    <NA>       0       0
#> 3    <NA>       0       0
#> 4       d       4       d
#> 5       e       5       e
# I think also that replace_char should convert numeric values to character automatically
# just like for factors so we don't have to explicitly do this:

convert_na_to(df, replace_num = 0,
              replace_char = "0",
              replace_fac = 0)
#>   letters numbers factors
#> 1       a       1       a
#> 2       0       0       0
#> 3       0       0       0
#> 4       d       4       d
#> 5       e       5       e
```
<sup>Created on 2022-08-10 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

datawizard v. 0.5.0

### PR DESCRIPTION
**Pull request by rempsyc**
Closes #214 
``` r
devtools::load_all("D:/OneDrive - UQAM/GitHub/forks/datawizard")
#> ℹ Loading datawizard
df <- data.frame(letters, numbers = 1:26, factors = as.factor(letters))[1:5,]
df[2:3,] <- NA
df
#>   letters numbers factors
#> 1       a       1       a
#> 2    <NA>      NA    <NA>
#> 3    <NA>      NA    <NA>
#> 4       d       4       d
#> 5       e       5       e

# Now works as expected for numeric replacements on character vectors
convert_na_to(df$letters, replacement = 0)
#> [1] "a" "0" "0" "d" "e"
convert_na_to(df, replace_num = 0,
              replace_char = 0,
              replace_fac = 0)
#>   letters numbers factors
#> 1       a       1       a
#> 2       0       0       0
#> 3       0       0       0
#> 4       d       4       d
#> 5       e       5       e

# Now works as expected when specifying a single replacement for multiple vector classes
convert_na_to(df, replacement = 0)
#>   letters numbers factors
#> 1       a       1       a
#> 2       0       0       0
#> 3       0       0       0
#> 4       d       4       d
#> 5       e       5       e

# It is possible to overwrite the global replacement if needed
convert_na_to(df, replacement = 0,
              replace_num = 7,
              replace_char = "Missing",
              replace_fac = "Factor")
#>   letters numbers factors
#> 1       a       1       a
#> 2 Missing       7  Factor
#> 3 Missing       7  Factor
#> 4       d       4       d
#> 5       e       5       e
```
<sup>Created on 2022-08-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
I have updated tests accordingly.

### CODE REVIEW

File: R/convert_na_to.R [hunk 1]
```diff
@@ -120,16 +120,17 @@ convert_na_to.factor <- function(x, replacement = NULL, verbose = TRUE, ...) {
 #' @rdname convert_na_to
 #' @export
 convert_na_to.character <- function(x, replacement = NULL, verbose = TRUE, ...) {
-  if (is_empty_object(replacement) || !is.character(replacement)) {
+  if (is_empty_object(replacement) || !is.character(replacement) & !is.numeric(replacement)) {
     if (isTRUE(verbose)) {
```
Comment (strengejacke) [inline R/convert_na_to.R:124]: Should this be `&` or `&&`?

File: tests/testthat/test-convert_na_to.R [hunk 2]
```diff
@@ -77,7 +77,7 @@ test_that("convert_na_to - numeric: arg 'replacement' must be of length one", {
 
 test_that("convert_na_to - character: returns original vector if 'replacement' not good", {
   expect_equal(
-    convert_na_to(c("a", "b", "c", NA), replacement = 1, verbose = FALSE),
+    convert_na_to(c("a", "b", "c", NA), replacement = mtcars, verbose = FALSE),
     c("a", "b", "c", NA)
```
Comment (strengejacke) [inline tests/testthat/test-convert_na_to.R:81]: Can you also test against the expected input (i.e. a numeric vector)?

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: R/convert_na_to.R (hunk 1 above)
Comment (rempsyc) [reply R/convert_na_to.R:124]: I have read about the booleans && and || but I still do not understand them very well (is it just generally more efficient/faster?). In any case, I have changed it so.

File: tests/testthat/test-convert_na_to.R (hunk 2 above)
Comment (rempsyc) [reply tests/testthat/test-convert_na_to.R:85]: Done, thanks for pointing that out

File: R/convert_na_to.R (hunk 1 above)
Comment (strengejacke) [reply R/convert_na_to.R:124]: See `?'&'`:

> & and && indicate logical AND and | and || indicate logical OR. The shorter forms performs elementwise comparisons in much the same way as arithmetic operators. The longer forms evaluates left to right, proceeding only until the result is determined. **The longer form is appropriate for programming control-flow and typically preferred in *if* clauses.**


``` r
# element-wise logical combination
c(TRUE, TRUE, FALSE) | c(FALSE, TRUE, FALSE)
#> [1]  TRUE  TRUE FALSE

# element-wise logical combination
FALSE | c(FALSE, TRUE, FALSE)
#> [1] FALSE  TRUE FALSE

# warning, length >= 3
c(TRUE, TRUE, FALSE) || c(FALSE, TRUE, FALSE)
#> Warning in c(TRUE, TRUE, FALSE) || c(FALSE, TRUE, FALSE): 'length(x) = 3 > 1' in
#> coercion to 'logical(1)'
#> [1] TRUE

# LHS and RHS are *one* logical expression
any(c(TRUE, TRUE, FALSE)) || all(c(FALSE, TRUE, FALSE))
#> [1] TRUE
```

<sup>Created on 2022-08-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Review by strengejacke — APPROVED** [review]

**Comment by strengejacke** [comment]
Thanks!

**Pull request merged**